collect_app/src/androidTest/java/org/odk/collect/android/FormNavigationTestCase.java

```diff
@@ -44,315 +44,60 @@
     private static final String FORMS_DIRECTORY = "/odk/forms/";
 
     @Test
-    public void test1() throws IOException {
+    public void formNavigationTestCase() throws IOException {
```

Comment:
It’s great to see this so much shorter. Thanks.

Reply:
Yes but now it's not possible to tell what case failed! I think this is a good opportunity for parameterized tests.

Reply:
@grzesiek2010 do you see what I mean by this? For example, change any of the `"-1, "` entries to `"-10"`. How will you know which case failed? 

You can read about parameterized tests at https://github.com/junit-team/junit4/wiki/parameterized-tests and @ggalmazor will surely have some good advice if you need.

Good to see a comment at the top! "it might be tricky" should be elaborated on -- what are the kinds of errors or regressions this verifies? Tests can be a big help when getting to know new code so it's a good opportunity to provide a little context. What do the lists of numbers represent? Why do you go backwards once at the end while navigating the form? Etc.

Reply:
I implemented changes using parameterized tests let me know if it is satisfying for you.

Reply:
But there is one problem when any test failed it displays a name of the last one...I'm not sure why it was ok when I tested an example from your link. I guess it's because we load a form and it's another task.